### DISCUSSION TITLE
Managing conditional transitive dependencies: How?

### DISCUSSION THREAD

**Discussion by McKillroy** [microsoft/vcpkg#24377] [Q&A] [answered]
I have a header only library I'd like to install with vcpkg manifest mode. 

It wraps an API and some additional functionality around another low level library, 
which could be exchanged against another low level library with similar functionality.
The conditional compile is done with a branching header, which conditionally (using macros) includes 
adapter headers which in turn include the low level libraries header.

On the user side these alternative low level libraries are chosen with a COMPONENT part in the 
find_package statement when using it:
```cmake
## like this when using low_level_library_ALPHA :
find_package(my_library COMPONENTS Alpha REQUIRED)
## or alternatively like so when using low_level_library_BETA :
find_package(my_library COMPONENTS Beta REQUIRED)
```
I also added the 'my_library' repo in a port inside a custom registry I'm using for users of this library.
I now wonder, How I could tell in the vcpkg manifest file of the user of my_library to chose the Alpha or Beta library 
transitive dependency when using my_library, (Alpha and Beta libs are also built with a vcpkg manifest and custom port and export a config file).
I have a feeling I should not make this a feature , which could probably do the job.
So - what would be the best way to achieve this? 
I'm not asking for any way, just - how would I solve this the vcpkg way?
Thanks
~Chris

**Comment by autoantwort** [comment, marked as answer]
> I have a feeling I should not make this a feature 

Why, because it is not a feature but an option between Alpha and Beta? So using Alpha and Beta or none of them is not possible?

**Reply by McKillroy** [reply]
I think it's because of this:
https://vcpkg.io/en/docs/maintainers/maintainer-guide.html
and especially this text there:
```text
Features
Do not use features to implement alternatives
```
So - exception from the rule? 
example user code:
```cmake
find_package(my_lib COMPONENTS Alpha CONFIG REQUIRED )
add_executable(main src/main.cpp)
target_link_libraries(main PUBLIC my_lib::my_lib_alpha)
```
On could say it's a feature, where one is required, and if both are installed only one can be used.
So ?

**Reply by autoantwort** [reply]
Ah ok, yeah you need https://github.com/microsoft/vcpkg/pull/22216, but it is not implemented yet. Maybe you can create two ports `mylib-alpha` and `mylib-beta`

**Reply by McKillroy** [reply]
Allright. So - maybe as an intermediate workaround - I'll make it a feature and transition it to whatever might solve the problem later.
Thanks for helping out.
